clang/lib/Driver/ToolChains/AIX.cpp

```diff
@@ -562,7 +562,7 @@ void AIX::addClangTargetOptions(
     CC1Args.push_back("-fno-sized-deallocation");
 
   if (Args.hasFlag(options::OPT_ferr_pragma_mc_func_aix,
-                   options::OPT_fno_err_pragma_mc_func_aix, false))
+                   options::OPT_fno_err_pragma_mc_func_aix, true))
```

Comment:
Since the preprocessor has a default value, can we avoid the cc1 option if the value here matches the default value inside the preprocessor, like:
```
if (Args.hasArg(options::OPT_fno_err_pragma_mc_func_aix))
   CC1Args.push_back("-fno-err-pragma-mc-func-aix");
```

(no need to pass `-ferr-pragma-mc-func-aix` as it is the default inside preprocessor.)

Reply:
Good point! Code is revised. 

I tried the suggested change
```
if (Args.hasArg(options::OPT_fno_err_pragma_mc_func_aix))
   CC1Args.push_back("-fno-err-pragma-mc-func-aix");
```
But it does not handle situations like `-fno-err-pragma-mc-func-aix -ferr-pragma-mc-func-aix` correctly. I think we still need to process both flags in the driver, but we only need to pass `-fno-err-pragma-mc-func-aix` to CC1. Did I miss something?

Reply:
Your code is the correct one. Thanks for pointing this out.